src/main/java/com/spikes2212/util/smartmotorcontrollers/TalonFXWrapper.java

```diff
@@ -0,0 +1,124 @@
+
+    public TalonFXWrapper(int deviceId, String canbus) {
+        talonFX = new TalonFX(deviceId, canbus);
+        closedLoopConfig = new Slot0Configs();
+        motorOutputConfigs = new MotorOutputConfigs();
+    }
+
+    public TalonFXWrapper(int deviceId, CANBus canbus) {
+        this(deviceId, canbus.getName());
+    }
+
+    public TalonFXWrapper(int deviceId) {
+        this(deviceId, "");
+    }
+
+    public void setIdleMode(NeutralModeValue neutralModeValue) {
+        talonFX.getConfigurator().apply(motorOutputConfigs.withNeutralMode(neutralModeValue));
+    }
+
+    public TalonFXConfigurator getConfigurator() {
+        return talonFX.getConfigurator();
+    }
+
+    @Override
+    public void configurePID(PIDSettings pidSettings) {
+        closedLoopConfig.kP = pidSettings.getkP();
+        closedLoopConfig.kI = pidSettings.getkI();
+        closedLoopConfig.kD = pidSettings.getkD();
+        talonFX.getConfigurator().apply(closedLoopConfig);
+    }
+
+    @Override
+    public void configureFF(FeedForwardSettings feedForwardSettings) {
+        closedLoopConfig.kS = feedForwardSettings.getkS();
+        closedLoopConfig.kV = feedForwardSettings.getkV();
+        closedLoopConfig.kA = feedForwardSettings.getkA();
+        closedLoopConfig.kG = feedForwardSettings.getkG();
+        talonFX.getConfigurator().apply(closedLoopConfig);
+    }
+
+    @Override
+    public void configureTrapezoid(TrapezoidProfileSettings trapezoidProfileSettings) {
+        MotionMagicConfigs config = new MotionMagicConfigs();
+        config.MotionMagicAcceleration = trapezoidProfileSettings.getAccelerationRate();
+        config.MotionMagicCruiseVelocity = trapezoidProfileSettings.getMaxVelocity();
+        config.MotionMagicJerk = trapezoidProfileSettings.getCurve();
+        talonFX.getConfigurator().apply(config);
+    }
+
+    @Override
+    public void pidSet(UnifiedControlMode controlMode, double setpoint, PIDSettings pidSettings,
+                       FeedForwardSettings feedForwardSettings, TrapezoidProfileSettings trapezoidProfileSettings) {
+        configurePID(pidSettings);
+        configureFF(feedForwardSettings);
+        configureTrapezoid(trapezoidProfileSettings);
+        ControlRequest request = switch (controlMode) {
+            case CURRENT -> new TorqueCurrentFOC(setpoint);
+            case PERCENT_OUTPUT -> new DutyCycleOut(setpoint);
+            case TRAPEZOID_PROFILE -> new MotionMagicDutyCycle(setpoint);
+            case MOTION_PROFILING -> throw new UnsupportedOperationException("Motion Profiling is not yet implemented in SpikesLib2!");
+            case VOLTAGE -> new VoltageOut(setpoint);
+            case VELOCITY -> new VelocityDutyCycle(setpoint);
+            case POSITION -> new PositionDutyCycle(setpoint);
+        };
+        talonFX.setControl(request);
+    }
+
+    public void follow(TalonFX master) {
+        talonFX.setControl(new Follower(master.getDeviceID(), false));
```

Comment:
test for this functionality.
use sparkMax and first set it inverted, and then follow.
do the same with talonFX.
are the results the same? are both motors inverted relative to the masters?